### PR TITLE
Fleshing out DB Schemas

### LINKED ISSUE #58: Figure out DB Schema
**User statement**
*As a developer, I would like a database schema to work with*

**Proposed solution**
Some tables that we would like in our database:
* Users (email, password, id)
* Events (user id, event type, timestamp, camera_id)
* Cameras (camera id, camera_name, user_id, status, mode)

**Development notes**
Add any other notes, context, or screenshots relevant to the ticket here

### PR DESCRIPTION
**Pull request by rudy-patel**
### What is it?
This PR adds the `camera` and `event` tables, along with a couple of placeholder API routes that add to these new tables. The [database](https://github.com/rudy-patel/imageAnalysisSystem/wiki/Database) page in the wiki has also been updated to reflect the changes.

### Screenshots
See the current table state below:
image

Closes #58.

### CODE REVIEW

File: enums/cameraEnums.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import enum
+
+class CameraMode(enum.Enum):
+    FACIAL_RECOGNITION = "FACIAL_RECOGNITION"
+    FAULT_DETECTION = "FAULT_DETECTION"
+
+class CameraStatus(enum.Enum):
+    ONLINE = "ONLINE"
+    OFFLINE = "OFFLINE"
```
Comment (rudy-patel) [inline enums/cameraEnums.py:9]: These enums store the camera mode/state. We can add/edit these later as we go.

File: enums/eventEnums.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import enum
+
+class EventType(enum.Enum):
+    FACIAL_MATCH_SUCCESS = "FACIAL_MATCH_SUCCESS"
+    FACIAL_MATCH_FAILED = "FACIAL_MATCH_FAILED"
+    FACIAL_MATCH_ERROR = "FACIAL_MATCH_ERROR"
+    FAULT_DETECT_SUCCESS = "FAULT_DETECT_SUCCESS"
+    NO_FAULT_DETECTED = "NO_FAULT_DETECTED"
+    IMAGE_CAPTURE_SUCCESS = "IMAGE_CAPTURE_SUCCESS"
+    IMAGE_CAPTURE_FAILED = "IMAGE_CAPTURE_FAILED"
```
Comment (rudy-patel) [inline enums/eventEnums.py:10]: These enums store the event types. We can add/edit these later as we go.

File: main.py [hunk 3]
```diff
@@ -9,12 +10,12 @@
...
 import boto3
-
-
+from enums.cameraEnums import CameraMode, CameraStatus
+from enums.eventEnums import EventType
 
 app = Flask(__name__)
 app.config['SECRET_KEY'] = 'imageanalysissystem'
-app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///database.db'
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///database/database.db'
```
Comment (rudy-patel) [inline main.py:18]: Created a new `database` directory to store our local file for the timebeing.

File: main.py [hunk 4]
```diff
@@ -27,14 +28,52 @@
...
 class User(db.Model, UserMixin):
     id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String(50))
-    email = db.Column(db.String(50), unique=True)
-    password = db.Column(db.String(80))
+    name = db.Column(db.String(50), nullable=False)
+    email = db.Column(db.String(50), unique=True, nullable=False)
+    password = db.Column(db.String(80), nullable=False)
+    events = db.relationship('Event', backref='user', lazy=True)
+    cameras = db.relationship('Camera', backref='user', lazy=True)
```
Comment (rudy-patel) [inline main.py:35]: These establish the foreign key relationships in the `event` and `camera` tables. This allows us to do something like `some_event.user` to get to that specific user related to that event.

More info here: https://flask-sqlalchemy.palletsprojects.com/en/2.x/models/

File: main.py [hunk 5]
```diff
@@ -27,14 +28,52 @@
...
+    camera_id = db.Column(db.Integer, db.ForeignKey('camera.id'), nullable=False)
+    type = db.Column(db.Enum(EventType), nullable=False)
+    timestamp = db.Column(db.DateTime, nullable=False)
+
+class Camera(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+    name = db.Column(db.String(50), nullable=False)
+    status = db.Column(db.Enum(CameraStatus), nullable=False)
+    mode = db.Column(db.Enum(CameraMode), nullable=False)
```
Comment (rudy-patel) [inline main.py:49]: Here are the new table models. Some things to note here are the **types** and **foreign key** constraints in some of the columns. I've also updated the database page on our wiki to reflect this.

File: main.py [hunk 6]
```diff
@@ -27,14 +28,52 @@
...
+@app.route("/addEvent")
+@login_required
+def testAddEvent():
+    
+    newEvent = Event(user_id=current_user.id, camera_id=1, type=EventType.FACIAL_MATCH_SUCCESS, timestamp=datetime.now())
+
+    db.session.add(newEvent)
+    db.session.commit()
+
+    return redirect(url_for('home'))
```
Comment (rudy-patel) [inline main.py:64]: This placeholder route adds a test event to the DB table.

File: main.py [hunk 7]
```diff
@@ -27,14 +28,52 @@
...
+@app.route("/addCamera")
+@login_required
+def testAddCamera():
+    
+    newCamera = Camera(user_id=current_user.id, name="Camera #" + str(current_user.id), status=CameraStatus.OFFLINE, mode=CameraMode.FACIAL_RECOGNITION)
+
+    db.session.add(newCamera)
+    db.session.commit()
+    
+    return redirect(url_for('home'))
```
Comment (rudy-patel) [inline main.py:75]: This placeholder route adds a test camera to the DB table.

**Review by b-stolte — APPROVED** [review]
Looks good, I'm working to get this integrated with AWS asap

**Review by kaden-dreger — APPROVED** [review]

**Pull request merged**